store/sqlstore/channel_store.go

```diff
@@ -1755,13 +1756,74 @@ func (s SqlChannelStore) AutocompleteInTeamForSearch(teamId string, userId strin
 			}
 		}
 
+		directChannels, err := s.autocompleteInTeamForSearchDirectMessages(userId, term)
+		if err != nil {
+			result.Err = err
+			return
+		}
+
+		channels = append(channels, directChannels...)
+
 		sort.Slice(channels, func(a, b int) bool {
 			return strings.ToLower(channels[a].DisplayName) < strings.ToLower(channels[b].DisplayName)
 		})
 		result.Data = &channels
 	})
 }
 
+func (s SqlChannelStore) autocompleteInTeamForSearchDirectMessages(userId string, term string) ([]*model.Channel, *model.AppError) {
+	queryFormat := `
+			SELECT
+				C.*
+			FROM
+				Channels AS C
+			JOIN
+				ChannelMembers AS CM ON CM.ChannelId = C.Id
+			WHERE
+			    C.Type = 'D'
+				AND CM.UserId = :UserId
+				AND C.Id IN (
```

Comment:
Can you explain the reason not to just `JOIN Users` in the top-level query?

Reply:
Same here, I was investigating and trying things.